pkg/mcp/server.go

```diff
@@ -103,15 +114,45 @@ func getApiClientFromContext(ctx context.Context) (promv1.API, error) {
 	return client, nil
 }
 
+// If there's an Authorization header, create a new RoundTripper
+// with the credentials from the header, otherwise return the
+// default client.
+func (m *apiClientLoaderMiddleware) getClient(header http.Header) promv1.API {
+	if header != nil && header.Get("Authorization") != "" {
+		var authType, secret string
+		if strings.Contains(header.Get("Authorization"), " ") {
+			authTypeCredentials := strings.Split(header.Get("Authorization"), " ")
+			if len(authTypeCredentials) != 2 {
+				m.logger.Error("Invalid Authorization header, falling back to default Prometheus client", "header", header.Get("Authorization"))
+				return m.defaultClient
+			}
+			authType = authTypeCredentials[0]
+			secret = authTypeCredentials[1]
+		} else {
+			m.logger.Debug("Assuming Bearer auth type for Authorization header with no type specified", "header", header.Get("Authorization"))
```

Comment:
Similar comment here -- a debug line to call out the assumption seems sane, but should we be concerned about logging secrets here?